internal/databases/postgres/backup_push_handler.go

```diff
@@ -75,226 +141,437 @@ func getDeltaConfig() (maxDeltas int, fromFull bool) {
 	return
 }
 
-func createAndPushBackup(bc *BackupConfig) {
-	folder := bc.uploader.UploadingFolder
+func (bc *BackupHandler) createAndPushBackup() {
+	folder := bc.workers.uploader.UploadingFolder
 	// TODO: AB: this subfolder switch look ugly.
 	// I think typed storage folders could be better (i.e. interface BasebackupStorageFolder, WalStorageFolder etc)
-	bc.uploader.UploadingFolder = folder.GetSubFolder(bc.backupsFolder)
+	bc.workers.uploader.UploadingFolder = folder.GetSubFolder(bc.arguments.backupsFolder)
+	tracelog.DebugLogger.Printf("Uploading folder: %s", bc.workers.uploader.UploadingFolder)
+
+	arguments := bc.arguments
+	meta := bc.curBackupInfo.meta
 	crypter := internal.ConfigureCrypter()
-	bundle := NewBundle(bc.archiveDirectory, crypter, bc.previousBackupSentinelDto.BackupStartLSN,
-		bc.previousBackupSentinelDto.Files, bc.forceIncremental, viper.GetInt64(internal.TarSizeThresholdSetting))
+	bc.workers.bundle = NewBundle(bc.pgInfo.pgDataDirectory, crypter, bc.prevBackupInfo.sentinelDto.BackupStartLSN,
+		bc.prevBackupInfo.sentinelDto.Files, arguments.forceIncremental,
+		viper.GetInt64(internal.TarSizeThresholdSetting))
 
-	var meta ExtendedMetadataDto
 	meta.StartTime = utility.TimeNowCrossPlatformUTC()
```

Comment:
These changes affect the local `meta` variable, not the `bc.curBackupInfo.meta`.

Reply:
Wanted to do a pointer, but I now changed this to use the original bc.curBackupInfo.meta.
But this will probably change as I am revising this part of the code as part of your other comment anyways.